x/gamm/keeper/pool_service.go

```diff
@@ -342,12 +342,19 @@ func (k Keeper) ExitSwapShareAmountIn(
 	if err != nil {
 		return sdk.Int{}, err
 	}
+
+	pool, err := k.GetPool(ctx, poolId)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+	swapFee := pool.GetSwapFee(ctx)
+
 	tokenOutAmount = exitCoins.AmountOf(tokenOutDenom)
 	for _, coin := range exitCoins {
 		if coin.Denom == tokenOutDenom {
 			continue
 		}
-		swapOut, err := k.SwapExactAmountIn(ctx, sender, poolId, coin, tokenOutDenom, sdk.ZeroInt())
+		swapOut, err := k.SwapExactAmountIn(ctx, sender, pool, coin, tokenOutDenom, sdk.ZeroInt(), swapFee)
```

Comment:
Wait, theres been an important abstraction issue thats changed here wrt swap fee.

I get why its being done here, but am concerned about:
- Whats the public/private guarantee
- Ensuring we keep dynamic swap fees possible.

The latter is still the case, if AMMs can "ignore" the swapFee arg.

Maybe we add a panic for now, if swapFee arg less than .5 ctx.SwapFee() ?

I think its good to merge if we add that safety check, and add to our mental checklist, that modules shouldn't import gamm, only swaprouter.

Reply:
I have the same concern and this is tracked here: https://github.com/osmosis-labs/osmosis/issues/3130

I've added the suggested check and tests for it in: [ace13f9](https://github.com/osmosis-labs/osmosis/pull/3764/commits/ace13f91ff1f2f79dffe0966fc2579ccd3af33e1)